parlai/agents/hred/hred.py

```diff
@@ -129,26 +129,15 @@ def batchify(self, obs_batch, sort=True):
         Store history vec as context_vec.
         """
         batch = super().batchify(obs_batch, sort)
-        batch["context_vec"], batch["hist_lens"] = self.parse_context_vec(batch)
-        return batch
-
-    def parse_context_vec(self, batch):
-        batch_context_vec = []
-        hist_lens = []
-        for i in range(len(batch["observations"])):
-            hist_len = len(batch["observations"][i]["context_vec"])
-            hist_lens.append(hist_len)
-            for j in range(hist_len):
-                context_vec = batch["observations"][i]["context_vec"][j]
-                batch_context_vec.append(torch.tensor(context_vec, device=self.device))
-
-        padded_context_vec = torch.nn.utils.rnn.pad_sequence(
-            batch_context_vec, batch_first=True
-        ).squeeze(1)
-        return (
-            padded_context_vec,
-            torch.tensor(hist_lens, dtype=torch.long, device=self.device),
+        # sum here is list concat, not addition
+        context_vec, hist_lens_ = self._pad_tensor(
```

Comment:
perhaps unrelated, but padded tensor returning the lengths as simply the lengths of the input lists is not super intuitive, especially when an optional argument is the pad token; ideally you'd want to return the lengths of the unpadded input lists

is this why you recompute `hist_lens` below?

Reply:
idk bro I'm just trying to get that one to work lol

Reply:
my b i didn't realize this was in the `hred` agent